crates/flux-refineck/src/type_env/place_ty.rs

```diff
@@ -661,7 +661,10 @@ where
             }
             TyKind::Indexed(BaseTy::Closure(def_id, upvar_tys, args), idx) => {
                 let upvar_tys = self.fold_field_at(upvar_tys, f);
-                Ty::indexed(BaseTy::Closure(*def_id, upvar_tys, args.clone()), idx.clone())
+                Ty::indexed(
+                    BaseTy::Closure(*def_id, upvar_tys, args.clone()), // TODO(RJ): Nico pls check!
```

Comment:
This didn't change after all

```suggestion
                    BaseTy::Closure(*def_id, upvar_tys, args.clone()),
```